packages/migrate/src/MigrateEngine.ts

```diff
@@ -351,9 +351,10 @@ export class MigrateEngine {
             this.rejectAll(err)
             reject(err)
           }
-          const engineMessage = this.lastError?.message || this.messages.join('\n')
+          const fallbackMessage = this.messages.join('\n')
+          const engineMessage = this.lastError?.message || fallbackMessage
           const handlePanic = () => {
-            const stackTrace = this.messages.join('\n')
+            const stackTrace = this.lastError?.backtrace || fallbackMessage
```

Comment:
Does the backtrace include the fallback message?

Do you have an example of output before and after this change, removing `fallbackMessage` is some cases and giving priority to `backtrace`?

Reply:
`backtrace` contains the actual stacktrace produced by the Rust panics. Example:

```text
           0: backtrace::capture::Backtrace::new
           1: migration_engine::set_panic_hook::{{closure}}
           2: std::panicking::rust_panic_with_hook
           3: std::panicking::begin_panic_handler::{{closure}}
           4: std::sys_common::backtrace::__rust_end_short_backtrace
           5: _rust_begin_unwind
           6: core::panicking::panic_fmt
           7: <migration_core::state::EngineState as migration_core::api::GenericApi>::debug_panic::{{closure}}
           8: <core::pin::Pin<P> as core::future::future::Future>::poll
           9: <futures_util::future::future::Then<Fut1,Fut2,F> as core::future::future::Future>::poll
          10: json_rpc_stdio::handle_stdin_next_line::{{closure}}::{{closure}}
          11: tokio::runtime::task::raw::poll
          12: tokio::runtime::scheduler::multi_thread::worker::Context::run_task
          13: tokio::runtime::task::raw::poll
          14: tokio::runtime::task::UnownedTask<S>::run
          15: std::sys_common::backtrace::__rust_begin_short_backtrace
          16: core::ops::function::FnOnce::call_once{{vtable.shim}}
          17: std::sys::unix::thread::Thread::new::thread_start
          18: __pthread_deallocate
```

Reply:
In most scenarios found in our error reporting dashboard, `fallbackMessage` is an empty string, but I've found an example where it contains a couple of logs from Rust:

```
Starting migration engine RPC server
The migration was not applied because it triggered warnings and the force flag was not passed.
[/Users/runner/.cargo/git/checkouts/quaint-9f01e008b9a89c14/1550f6c/src/connector/result_set/result_row.rs:59:64] index out of bounds: the len is 1 but the index is 1
``` 

Reply:
Do you think it makes sense then to concatenate the two so we have at least the same previous output? Or does that not matter at all in this case?

Reply:
Considering this discovery, a better alternative would be including both `this.messages.join('\n')` and `this.lastError?.backtrace` in `RustPanic.stacktrace`.